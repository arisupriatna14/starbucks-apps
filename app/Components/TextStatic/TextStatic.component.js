import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

const TextStatic = (props) => (
  <Text style={props.styleText}>
    {props.text}
  </Text>
);

export default TextStatic;

TextStatic.propTypes = {
  styleText: PropTypes.instanceOf(Object),
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
};