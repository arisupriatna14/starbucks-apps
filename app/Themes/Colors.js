const Colors = {
  MINE_SHAFT: '#333333',
  SCORPION: '#5c5c5c',
  WHITE: '#ffffff',
  SALEM: '#088f4a',
  SAFFRON: '#f1c63f',
  BLACK: '#000000',
  TUNDORA: '#414141'
};

export { Colors };
