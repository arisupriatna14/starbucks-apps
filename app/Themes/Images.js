const Images = {
  logoStarbucks: require('../Assets/Images/logo_starbuck.png'),
  icCoffeOne: require('../Assets/Images/ic_coffe_1.png'),
  icCoffeTwo: require('../Assets/Images/ic_coffe_2.png'),
  icCoffeThree: require('../Assets/Images/ic_coffe_3.png'),
  icBagBlack: require('../Assets/Images/ic_bag_black.png'),
  icBagWhite: require('../Assets/Images/ic_bag_white.png'),
  icSearch: require('../Assets/Images/ic_search.png'),
  icBack: require('../Assets/Images/ic_back.png'),
  coffeeBag: require('../Assets/Images/ic_coffe_bag.png')
};

export { Images };
