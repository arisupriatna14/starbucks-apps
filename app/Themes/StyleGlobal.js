/**
 * StylesGlobal
 * 
 * @param {number} flex - value is 1 - 6
 * @param {string} flexDirection - column, column-reverse, row, row-reverse
 * @param {number} marginHorizontal
 * @param {number} marginVertical
 * @param {number} marginTop
 * @param {number} marginBottom
 * @param {number} marginLeft
 * @param {number} marginRight
 * @param {number} top
 * @param {number} bottom
 * @param {number} left
 * @param {number} right
 * @param {number} sizeImg - width and height
 * @param {number} sizeLayout - width and height
 * @param {number and string} container - flex and backgroundColor
 * @param {object} viewCenter
 */

const StylesGlobal = {
  flex: value => ({
    flex: value
  }),
  flexDirection: value => ({
    flexDirection: value
  }),
  marginHorizontal: value => ({
    marginHorizontal: value
  }),
  marginVertical: value => ({
    marginVertical: value
  }),
  marginTop: value => ({
    marginTop: value
  }),
  marginBottom: value => ({
    marginBottom: value
  }),
  marginLeft: value => ({
    marginLeft: value
  }),
  marginRight: value => ({
    marginRight: value
  }),
  margin: value => ({
    margin: value
  }),
  top: value => ({
    top: value
  }),
  bottom: value => ({
    bottom: value
  }),
  left: value => ({
    left: value
  }),
  right: value => ({
    right: value
  }),
  sizeImg: (width, height) => ({
    width,
    height
  }),
  sizeLayout: (width, height) => ({
    width,
    height
  }),
  viewCenter: {
    justifyContent: 'center',
    alignItems: 'center'
  }
};

export { StylesGlobal };
