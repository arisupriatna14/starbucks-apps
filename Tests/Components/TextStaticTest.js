import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TextStatic from '../../app/Components/TextStatic/TextStatic.component';
import { Colors } from '../../app/Themes/Colors';

test('TextStatic component renders correctly if text prop no set', () => {
  const tree = renderer.create(<TextStatic />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextStatic component renders correctly if text prop is string', () => {
  const tree = renderer.create(<TextStatic text="Hello world" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextStatic component renders correctly if text prop is number', () => {
  const tree = renderer.create(<TextStatic text={100} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextStatic component renders correctly if text and styleText is set', () => {
  const tree = renderer.create(
    <TextStatic
      text="Hello world"
      styleText={{
        color: Colors.WHITE
      }}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('TextStatic', () => {
  it('Component will renders correctly if text with prop types is number', () => {
    const testRenderer = renderer.create(<TextStatic text={50} />);
    const testInstance = testRenderer.root.findByType(TextStatic);
    expect(testInstance.props.text).toEqual(50);
    expect(typeof testInstance.props.text).toEqual('number');
  });

  it('Component will renders correctly if text with prop types is string', () => {
    const testRenderer = renderer.create(<TextStatic text="Title Button" />);
    const testInstance = testRenderer.root.findByType(TextStatic);
    const props = Object.keys(testInstance.props);

    expect(props).toEqual(['text']);
    expect(typeof testInstance.props.text).toEqual('string');
    expect(testInstance.props.text).toEqual('Title Button');
  });

  it('Component will renders correctly if text and styles set', () => {
    const testRenderer = renderer.create(
      <TextStatic
        text="Hello world"
        styleText={{
          color: Colors.WHITE
        }}
      />
    );

    const testInstance = testRenderer.root;

    expect(typeof testInstance.props.styleText).toBe('object');
    expect(typeof testInstance.props.text).toBe('string');
    expect(Object.keys(testInstance.props)).toEqual(['text', 'styleText']);
    expect(Object.values(testInstance.props)).toEqual(['Hello world', { color: Colors.WHITE }]);
  });
});
