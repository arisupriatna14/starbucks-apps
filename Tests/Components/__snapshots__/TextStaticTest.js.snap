// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextStatic component renders correctly if text and styleText is set 1`] = `
<Text
  style={
    Object {
      "color": "#ffffff",
    }
  }
>
  Hello world
</Text>
`;

exports[`TextStatic component renders correctly if text prop is number 1`] = `
<Text>
  100
</Text>
`;

exports[`TextStatic component renders correctly if text prop is string 1`] = `
<Text>
  Hello world
</Text>
`;

exports[`TextStatic component renders correctly if text prop no set 1`] = `<Text />`;
